### PR TITLE
feat(new-primary-school):  Implement parent page

### PR DESCRIPTION
**Pull request by birkirkristmunds**
[TS-799](https://dit-iceland.atlassian.net/browse/TS-799) Implement parent page

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 1]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
-                  value={parent.phoneNumber}
-                />
-              </GridColumn>
-            </GridRow>
+                <GridColumn span={['12/12', '12/12', '12/12', '5/12']}>
+                  <DataValue
+                    label={formatMessage(
+                      newPrimarySchoolMessages.confirm.phoneNumber,
+                    )}
+                    value={parent.phoneNumber}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:97]: Fixed

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 2]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
+            <Stack space={2}>
+              <GridRow rowGap={2}>
+                <GridColumn span={['12/12', '12/12', '12/12', '12/12']}>
+                  <Label>
+                    {formatMessage(newPrimarySchoolMessages.confirm.parents)}{' '}
+                    {+parentIndex}
+                  </Label>
+                </GridColumn>
+              </GridRow>
+              <GridRow marginTop={3}>
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:47]: Fixed

File: libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm.ts [hunk 3]
```diff
@@ -259,13 +272,19 @@ export const NewPrimarySchoolForm: Form = buildForm({
                   dataTestId: 'email2',
                   id: 'parent2.email',
                   variant: 'email',
+                  required: true,
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm.ts:275]: yes absolutely

File: libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm.ts [hunk 4]
```diff
@@ -259,13 +272,19 @@ export const NewPrimarySchoolForm: Form = buildForm({
...
+                  condition: (answers, externalData) =>
+                    hasOtherParent(answers, externalData),
                 }),
                 buildPhoneField({
                   width: 'half',
                   title: newPrimarySchoolMessages.childrenNParents.phoneNumber,
                   id: 'parent2.phoneNumber',
                   dataTestId: 'phone2',
                   placeholder: '000-0000',
+                  required: true,
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm.ts:285]: yes absolutely

File: libs/application/templates/new-primary-school/src/lib/dataSchema.ts [hunk 5]
```diff
@@ -41,5 +75,3 @@ export const dataSchema = z.object({
       params: errorMessages.relativesRequired,
     }),
 })
-
-export type SchemaFormValues = z.infer<typeof dataSchema>
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/lib/dataSchema.ts:45]: wow, that should not be deleted

[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/lib/dataSchema.ts:45]: added back

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 6]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
-                  value={parent.address.streetAddress}
-                />
-              </GridColumn>
+            <Stack space={2}>
+              <GridRow rowGap={2}>
+                <GridColumn span={['12/12', '12/12', '12/12', '12/12']}>
+                  <Label>
+                    {formatMessage(newPrimarySchoolMessages.confirm.parents)}{' '}
+                    {+parentIndex}
+                  </Label>
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:44]: fixed

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 7]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
+                    value={parent.address.city}
+                  />
+                </GridColumn>
+              </GridRow>
+              <GridRow rowGap={2}>
+                <GridColumn span={['12/12', '12/12', '12/12', '5/12']}>
+                  <DataValue
+                    label={formatMessage(
+                      newPrimarySchoolMessages.confirm.email,
+                    )}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:87]: Moved to shared and changed in both places

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 8]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
+                    value={formatKennitala(parent.nationalId)}
+                  />
+                </GridColumn>
+              </GridRow>
+              <GridRow rowGap={2}>
+                <GridColumn span={['12/12', '12/12', '12/12', '5/12']}>
+                  <DataValue
+                    label={formatMessage(
+                      newPrimarySchoolMessages.confirm.address,
+                    )}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:68]: Moved to shared and changed in both places

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 9]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
-                <DataValue
-                  label={formatMessage(newPrimarySchoolMessages.confirm.email)}
-                  value={parent.email}
-                />
-              </GridColumn>
+                <GridColumn span={['12/12', '12/12', '12/12', '5/12']}>
+                  <DataValue
+                    label={formatMessage(
+                      newPrimarySchoolMessages.confirm.municipality,
+                    )}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx:77]: Moved to shared and changed in both places

File: libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx [hunk 10]
```diff
@@ -19,83 +19,86 @@ export const Parents = ({
...
+                  <DataValue
+                    label={formatMessage(newPrimarySchoolMessages.confirm.name)}
+                    value={parent.fullName}
+                  />
+                </GridColumn>
+                <GridColumn span={['12/12', '12/12', '12/12', '5/12']}>
+                  <DataValue
+                    label={formatMessage(
+                      newPrimarySchoolMessages.confirm.nationalId,
+                    )}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/new-primary-school/src/fields/Review/review-groups/Parents.tsx]: Moved to shared and changed in both places

**Review by veronikasif — APPROVED** [review]
lgtm 👍

**Pull request merged**